shipyard_blueprints/s3/integration/int.py

```diff
@@ -0,0 +1,245 @@
+import os
+import subprocess
+import pytest
+import shutil
+from dotenv import load_dotenv, find_dotenv
+
+load_dotenv(find_dotenv())
+
+
+@pytest.fixture(scope="module")
+def aws_credentials():
+    return {
+        "AWS_ACCESS_KEY_ID": os.getenv("AWS_ACCESS_KEY_ID"),
+        "AWS_SECRET_ACCESS_KEY": os.getenv("AWS_SECRET_ACCESS_KEY"),
+        "REGION": os.getenv("REGION"),
+        "BUCKET": os.getenv("BUCKET"),
+        "S3_FOLDER": os.getenv("S3_FOLDER"),
+    }
+
+
+def test_upload_download_integration(aws_credentials):
+    if not all(aws_credentials.values()):
+        pytest.skip("AWS credentials are not properly configured.")
+
+    # Upload
+    upload_command = [
+        "python3",
+        "./shipyard_s3/cli/upload.py",
+        "--aws-access-key-id",
+        aws_credentials["AWS_ACCESS_KEY_ID"],
+        "--aws-secret-access-key",
+        aws_credentials["AWS_SECRET_ACCESS_KEY"],
+        "--aws-default-region",
+        aws_credentials["REGION"],
+        "--bucket-name",
+        aws_credentials["BUCKET"],
+        "--source-file-name-match-type",
+        "exact_match",
+        "--source-file-name",
+        "s3.csv",
+        "--destination-folder-name",
+        aws_credentials["S3_FOLDER"],
+    ]
+    subprocess.run(upload_command, check=True)
+
+    # Download
+    download_command = [
+        "python3",
+        "./shipyard_s3/cli/download.py",
+        "--aws-access-key-id",
+        aws_credentials["AWS_ACCESS_KEY_ID"],
+        "--aws-secret-access-key",
+        aws_credentials["AWS_SECRET_ACCESS_KEY"],
+        "--aws-default-region",
+        aws_credentials["REGION"],
+        "--bucket-name",
+        aws_credentials["BUCKET"],
+        "--source-folder-name",
+        aws_credentials["S3_FOLDER"],
+        "--source-file-name",
+        "s3.csv",
+        "--source-file-name-match-type",
+        "exact_match",
+        "--destination-file-name",
+        "new_s3.csv",
+    ]
+    subprocess.run(download_command, check=True)
+
+    # Check if files match
+    original_rows = sum(1 for _ in open("s3.csv"))
+    new_rows = sum(1 for _ in open("new_s3.csv"))
```

Comment:
not something that needs to be changed but would also be interesting to check for would be file size. Row count is enough for testing purposes.